thinc/layers/concatenate.py

```diff
@@ -98,33 +104,33 @@ def backprop(d_output: Ragged) -> InT:
     return output, backprop
 
 
-def _list_forward(model: Model[InT, List[Array2d]], X, Ys, callbacks, is_train: bool):
-    lengths = model.ops.asarray1i([len(x) for x in X])
-    Ys = [model.ops.xp.concatenate(Y, axis=0) for Y in Ys]
-    widths = [Y.shape[1] for Y in Ys]
-    out_array = model.ops.xp.hstack(Ys)
-    output = model.ops.unflatten(out_array, lengths)
-
+def _list_forward(
+    model: Model[InT, OutT], X, Ys: List, callbacks, is_train: bool
+) -> Tuple[List[Array2d], Callable]:
```

Comment:
Does it matter whether this is returning `Sequence[Array2d]` or `List[Array2d]` ?

Reply:
I would say no, because the return value is cast to `OutT` in the `forward()` method and `OutT` declares `Sequence[Array2d]`.

Reply:
Then perhaps we can type it with `Sequence`? To me it would help link the types of the different private methods up to the `Union` of types from the original output type. These are really nitpicking aestethics details though ;-)